xcdat/averager.py

```diff
@@ -353,11 +377,13 @@ def _averager(xvariable, weights, axis):
     return wavg
 
 
-def spatial_average(xvariable,
-                    lat_bounds=None,
-                    lon_bounds=None,
-                    axis=['lat', 'lon'],
-                    weight='weighted'):
+def spatial_average(
+    xvariable: xr.DataArray,
+    axis: Union[List[SupportedAxes], SupportedAxes],
+    lat_bounds: Optional[RegionAxisBounds] = None,
+    lon_bounds: Optional[RegionAxisBounds] = None,
+    is_weighted: bool = True,
```

Comment:
`is_weighted` boolean instead of `weighted` as a str or dataarray (this wasn't supported yet).